core/src/main/java/hudson/util/DescribableList.java

```diff
@@ -270,10 +270,9 @@ public void marshal(Object source, HierarchicalStreamWriter writer, MarshallingC
         }
 
         public Object unmarshal(HierarchicalStreamReader reader, UnmarshallingContext context) {
-            CopyOnWriteList core = copyOnWriteListConverter.unmarshal(reader, context);
-
             try {
-                DescribableList r = (DescribableList)context.getRequiredType().newInstance();
+                DescribableList r = (DescribableList) context.getRequiredType().asSubclass(DescribableList.class).newInstance();
```

Comment:
`asSubclass` ensures that even if the problem occurs, we will get a `ClassCastException`, which is easier to diagnose than a `NoSuchMethodException`.